File: go/basics/fizzbuzz.go
// Print FizzBuzz for 1 through 100, choosing the label in a switch with no condition.
// Run: go run fizzbuzz.go

package main

import "fmt"

func label(n int) string {
	switch {
	case n%15 == 0:
		return "FizzBuzz"
	case n%3 == 0:
		return "Fizz"
	case n%5 == 0:
		return "Buzz"
	default:
		return fmt.Sprint(n)
	}
}

func main() {
	for n := 1; n <= 100; n++ {
		fmt.Println(label(n))
	}
}

File: r/basics/fizzbuzz.R
# Print FizzBuzz for 1 through 100, choosing the label from the remainders.
# Run: Rscript fizzbuzz.R

label <- function(n) {
  if (n %% 15 == 0) {
    "FizzBuzz"
  } else if (n %% 3 == 0) {
    "Fizz"
  } else if (n %% 5 == 0) {
    "Buzz"
  } else {
    as.character(n)
  }
}

for (n in 1:100) {
  cat(label(n), "\n", sep = "")
}
